### PR TITLE
Invalidate all user sessions

### PR DESCRIPTION
**Pull request by fede1608**
## Summary

- POST /users/sessions/invalidate_all implemented
- Tests added

## Trello Card

[https://trello.com/c/6kUkaHXv/35-deshabilitar-todas-las-sesiones](https://trello.com/c/6kUkaHXv/35-deshabilitar-todas-las-sesiones)

### CODE REVIEW

File: migrations/migrations/20181023175544-users-model.js [hunk 1]
```diff
@@ -32,6 +32,9 @@ module.exports = {
         defaultValue: false,
         allowNull: false
       },
+      sessionInvalidate: {
```
Comment (mpizzagalli) [inline migrations/migrations/20181023175544-users-model.js]: Remember the idea of migrations is to add a new migration each time you make some change in the db schema

File: app/controllers/users.js [hunk 2]
```diff
@@ -138,5 +138,15 @@ module.exports = {
           });
       }
     });
+  },
+  invalidateAllSessions(req, res, next) {
+    return User.update(
+      { sessionInvalidate: Math.floor(Date.now() / 1000) },
+      { returning: true, where: { email: req.user.email } }
+    )
+      .then(function([rowsUpdate, [userUpdated]]) {
```
Comment (mpizzagalli) [inline app/controllers/users.js]: use fat arrow functions

File: app/utils.js [hunk 3]
```diff
@@ -0,0 +1,17 @@
...
+const logger = require('./logger');
+
+module.exports = {
+  isRevokedCallback(req, payload, done) {
+    const issuer = payload.iat;
+    const tokenId = payload.jti;
+    User.findOne({ where: { id: payload.id } }).then(user => {
+      if (user.sessionInvalidate && user.sessionInvalidate >= issuer) {
+        logger.error('Token Invalidated');
+        done(new Error('Token Invalidated'));
```
Comment (mpizzagalli) [inline app/utils.js:11]: Is it possible to use some of our errors of `app/errors.js` here?

**Review by mpizzagalli — CHANGES_REQUESTED** [review]

File: app/controllers/users.js (hunk 2 above)
Comment (fede1608) [reply app/controllers/users.js]: fixed

File: app/utils.js (hunk 3 above)
Comment (fede1608) [reply app/utils.js:11]: Fixed in #13

**Review by mpizzagalli — APPROVED** [review]

**Pull request merged**